### PR TITLE
tests: migrate to use new client and server API

### PR DESCRIPTION
**Pull request by aarnphm**
## What does this PR address?

Migrate current `bentoml.testing` to use the recent server and client API.

Remove broken pandas series tests for now, will investigate on a
different tickets.

depends on #3843, #3844, #3845, #3846

- In addition to the test improvement, this PR also introduces a `async_request` and `request` to `HTTPClient`. This enables users to send request other that `POST` to the server.

This is useful when BentoService is mounted with a fastapi or other ASGI app.

`client.async_request` is a thin wrapper for `aiohttp.ClientSession.request`.

- all attributes that ``aio.ClientSession.post`` accepts that has prefix of `_http_*` can also be passed from `client.<endpoint>`

```python

client.predict(np.array([1,2]), _http_timeout=90, _http_headers={"Content-Type": "application/json"})

```

Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/server.py [hunk 1]
```diff
@@ -87,14 +88,20 @@ def __init__(
...
+        self.host = host
         self.port = port
         self.start = self._create_startmanager()
 
-    def _create_startmanager(server):  # type: ignore # not calling self self
+    def _create_startmanager(self):
         class _StartManager:
             def __init__(
-                self, blocking: bool = False, env: dict[str, str] | None = None
+                __start_self,
```
Comment (aarnphm) [inline src/bentoml/server.py:98]: apparently pyright accepts this behaviour 🤷

**Comment by sauyon** [comment]
Should we circle back to this one later?

**Review by sauyon — COMMENTED** [review]
Apparently I had some pending comments.

**Comment by sauyon** [comment]
Going to close this one, it's not a big priority.

**Pull request closed without merging**